app/scripts/modules/titus/src/serverGroup/configure/wizard/pages/disruptionBudget/JobDisruptionBudget.tsx

```diff
@@ -99,6 +101,15 @@ export class JobDisruptionBudget extends React.Component<IJobDisruptionBudgetPro
     return options.find((o) => !!disruptionBudget[o.field]);
   }
 
+  componentDidUpdate = (oldProps: IJobDisruptionBudgetProps) => {
+    if (
+      this.props.onStageChange &&
+      !isEqual(oldProps.formik.values.disruptionBudget, this.props.formik.values.disruptionBudget)
+    ) {
+      this.props.onStageChange(this.props.formik.values.disruptionBudget);
```

Comment:
Does this happen on every change of the form or only on submit?

Reply:
Only when the disruption budget is updated. I was having some trouble syncing the underlying model to what was rendered between the angular parent and react child

Reply:
I was just concerned that it will re-render every time the user types. You can ignore if it's not the case or if it's not a big deal